overlord/ifacestate/ifacestate.go

```diff
@@ -238,7 +238,12 @@ func initialConnectAttributes(st *state.State, plugSnap string, plugName string,
 		return nil, nil, err
 	}
 
-	addImplicitSlots(snapInfo)
+	hotplugSlots, err := getHotplugSlots(st)
+	if err != nil {
+		return nil, nil, err
+	}
+	addImplicitSlots(snapInfo, hotplugSlots)
```

Comment:
If you follow my advice this function should return an error, and we should handle it here